app/lib/features/events/widgets/event_date_widget.dart

```diff
@@ -0,0 +1,51 @@
+import 'package:acter/common/utils/utils.dart';
+import 'package:acter/features/events/event_utils/event_utils.dart';
+import 'package:acter/features/events/providers/event_providers.dart';
+import 'package:acter_flutter_sdk/acter_flutter_sdk_ffi.dart';
+import 'package:flutter/material.dart';
+
+class EventDateWidget extends StatelessWidget {
+  final CalendarEvent calendarEvent;
+  final double size;
+
+  const EventDateWidget({
+    super.key,
+    required this.calendarEvent,
+    this.size = 70,
```

Comment:
as the chat boosting has just shown, passing big objects around can quickly slow down generating the render-tree ... we might want to consider using eventId strings and providers primarily ... but I don't think we'll see enough events for this to become an issue soon. but something to keep in mind.

Reply:
Agree but prefer to do in separate PR so that this big PR not get stretched out much.